tmva/pymva/src/PyMethodBase.cxx

```diff
@@ -136,22 +136,26 @@ void PyMethodBase::PyInitialize()
       _import_array();
    }
 
+   // note fMain is a borrowed reference
    fMain = PyImport_AddModule("__main__");
    if (!fMain) {
       Log << kFATAL << "Can't import __main__" << Endl;
       Log << Endl;
    }
+   Py_INCREF(fMain);
 
    fGlobalNS = PyModule_GetDict(fMain);
    if (!fGlobalNS) {
       Log << kFATAL << "Can't init global namespace" << Endl;
       Log << Endl;
    }
+   // no need to increase reference for fGlobalNS, since it is linked to fMain?
```

Comment:
This is probably right, but anyway it would be good to INCREF it if you intend to keep it, since you got it as a borrowed reference.

Reply:
Yes, I could do that, but I was thinking that fGlobalNS is linked to fMain and as far aa fMain is alive, fGlobalNS will be. Do you think is safer increasing the count and then decrease it in PyFinalize() ? At the moment this is not done

Reply:
I think it's better practice. If you got a borrowed reference but you intend to keep the object, you INCREF. This way you are protected in case (even if here it shouldn't happen) `fMain` is destroyed but you still need `fGlobalNS`.

Reply:
The PR is updated with your suggestion. Thank you !